### PR TITLE
Add support for mat3 to shader compiler

### PR DESCRIPTION
**Pull request by bghgary**

### CODE REVIEW

File: Plugins/NativeEngine/Source/NativeEngine.cpp [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ namespace Babylon
 
         assert(matrix.size() == size * size);
 
-        if constexpr (size < 4)
+        if constexpr (size < 3)
```
Comment (docEdub) [inline Plugins/NativeEngine/Source/NativeEngine.cpp:1064]: Is this right?

File: Plugins/NativeEngine/Source/NativeEngine.cpp [hunk 2]
```diff
@@ -1061,7 +1061,7 @@ namespace Babylon
...
         {
             std::array<float, 16> matrixValues{};
```
Comment (PolygonalSun) [inline Plugins/NativeEngine/Source/NativeEngine.cpp:1066]: Do we have to worry about any sort of side effects by storing a 3x3 matrix across a size 16 array?  I'm assuming that the untouched elements of the array start as 0 but I wanted to make sure.

File: Plugins/NativeEngine/Source/NativeEngine.cpp (hunk 1 above)
Comment (bghgary) [reply Plugins/NativeEngine/Source/NativeEngine.cpp:1064]: Yup. This is needed for mat3.

File: Plugins/NativeEngine/Source/NativeEngine.cpp (hunk 2 above)
Comment (bghgary) [reply Plugins/NativeEngine/Source/NativeEngine.cpp:1066]: The `{}` will initialize the array to the default value of `float` which is `0.0f`

Comment (PolygonalSun) [reply Plugins/NativeEngine/Source/NativeEngine.cpp:1066]: Disregard, I misread this.

**Review by docEdub — APPROVED** [review]

**Review by PolygonalSun — APPROVED** [review]

**Pull request merged**